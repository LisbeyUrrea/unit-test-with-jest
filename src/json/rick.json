{
    "id":2,
    "name":"Rick Sanchez",
    "status": "Alive",
    "species": "Human",
    "gender": "Male"
}