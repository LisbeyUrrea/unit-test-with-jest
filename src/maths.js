export const suma = (numeroUo, numeroDos) => numeroUo + numeroDos;

export const resta = (numeroUo, numeroDos) => numeroUo - numeroDos;

export const multiplicacion = (numeroUo, numeroDos) => numeroUo * numeroDos;

export const division = (numeroUo, numeroDos) => numeroUo / numeroDos;


