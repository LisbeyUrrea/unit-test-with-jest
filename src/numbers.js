export const suma = (numeroUno, numeroDos) => numeroUno + numeroDos;