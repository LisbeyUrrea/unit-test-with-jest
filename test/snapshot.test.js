import { getCharacter } from '../src/snapshot'
import rick from '../src/json/rick.json'

describe('SnapShop - Instantanesas', () => {

    test('primer SnapShop ', () => {
        expect(getCharacter(rick)).toMatchSnapshot()
    })

    test('varias exepciones ', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }

        expect(user).toMatchSnapshot({ createAt: expect.any(Date), id: expect.any(Number) })
    })

    test('Exepsion de codigo', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Lisbey Urrea"
        }
        expect(user).toMatchSnapshot({ id: expect.any(Number) })
    })
})