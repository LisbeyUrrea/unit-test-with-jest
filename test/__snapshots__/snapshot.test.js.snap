// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShop - Instantanesas Exepsion de codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Lisbey Urrea",
}
`;

exports[`SnapShop - Instantanesas deberia fallar la instantanea  1`] = `
Object {
  "createAt": Any<Date>,
  "id": Any<Number>,
}
`;

exports[`SnapShop - Instantanesas primer SnapShop  1`] = `
Object {
  "gender": "Male",
  "id": 2,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
